src/hooks/hook_area02.nss

```diff
@@ -15,8 +15,20 @@ void main()
     // Don't run this event if the exiting object is a PC that is about to be
     // booted.
     object oPC = GetExitingObject();
-    if (GetIsPC(oPC) && GetLocalInt(oPC, LOGIN_BOOT))
-        return;
+
+    if (HasListObject(OBJECT_SELF, oPC, AREA_ROSTER))
+    {
+        if (GetLocalInt(oPC, LOGIN_BOOT))
```

Comment:
The login boot check should happen before this block. This check helps to ensure that a PC that should be booted OnClientEnter doesn't interact with any other events.

Reply:
It's out of the conditional.  New commit on the way.